_benchmark/creating-a-custom-workload.md

```diff
@@ -0,0 +1,374 @@
+---
+layout: default
+title: Creating a custom workload
+nav_order: 10
+has_children: false
+---
+
+# Creating a custom workload
+
+OpenSearch Benchmark comes packaged with a set of [workloads](https://github.com/opensearch-project/opensearch-benchmark-workloads) you can use to benchmark data from your cluster. However, if you want to create workload tailored to your own data, create a custom workload using one of the following options:
+
+- [Creating a workload from an existing cluster](#creating-a-workload-from-an-existing-cluster)
+- [Creating a workload without an existing cluster](#creating-a-workload-without-an-existing-cluster)
+
+## Creating a workload from an existing cluster
+
+If you already have an OpenSearch cluster with indexed data, use the following steps to create a custom workload for your cluster.
+
+### Prerequisites
+
+Before creating a custom workload, make sure you have the following prerequisites:
+
+- An OpenSearch cluster with an index that contains over 1000 documents. If your cluster's index does not contain at least 1000 documents, the workload can still run tests, however, you will not be able to run workloads using `--test-mode`.
+- Make sure you have the correct permissions to access your OpenSearch cluster. For more information about cluster permissions, see [Permissions]({{site.url}}{{site.baseurl}}/security/access-control/permissions/).
+
+### Customizing the workload
+
+To begin creating your custom workload, use the `opensearch-benchmark create-workload` command. 
+
+```
+opensearch-benchmark create-workload \
+--workload="<WORKLOAD NAME>" \
+--target-hosts="<CLUSTER ENDPOINT>" \
+--client-options="basic_auth_user:'<USERNAME>',basic_auth_password:'<PASSWORD>'" \
+--indices="<INDICES TO GENERATE WORKLOAD FROM>" \
+--output-path="<LOCAL DIRECTORY PATH TO STORE WORKLOAD>"
+```
+
+Replace the following options in the preceding example with information specific to your existing cluster:
+
+- `--workload`: A custom name for your custom workload.
+- `--target-hosts:` A comma-separated list of host:port pairs that workload targets.
```

Comment:
Instead of "that workload targets", could this be updated to something such as "cluster to extract data from"?